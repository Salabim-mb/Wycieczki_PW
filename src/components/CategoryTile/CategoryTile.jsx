import React from 'react';
import { Link } from 'react-router-dom'
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';


const CategoryTile = ({ title, desc, cover, link }) => (
    <Card >
        <CardActionArea component={Link} to={link}>
            <CardMedia
                image={cover}
                title={title}
            />
            <CardContent>
                <Typography gutterBottom variant="h5" component="h2">
                    {title}
                </Typography>
                <Typography variant="body2" color="textSecondary" component="p">
                    {desc}
                </Typography>
            </CardContent>
        </CardActionArea>
        <CardActions>
            <Button component={Link} to={link} size="small" color="primary">
                Learn More
        </Button>
        </CardActions>
    </Card>
);

export default CategoryTile