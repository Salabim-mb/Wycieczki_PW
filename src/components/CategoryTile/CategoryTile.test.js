import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import CategoryTile from './CategoryTile';

import logo from '../../logo.svg'
// TRZEBA MOCK JESZCZE MATERIALA 
describe('CategoryTile', () => {
    let props;
    beforeEach(() => {
        props = {

            title: 'Geometria przestrzenna',
            cover: logo,
            desc: 'Opis jakiś tam coś tam bla bla',
            link: '/liceum/kolorowanki/1',
        };
    });
    it('should match snapshot', () => {
        const { container } = render(
            <BrowserRouter>
                <CategoryTile {...props} />
            </BrowserRouter>
        );

        expect(container).toMatchSnapshot();
    });

});