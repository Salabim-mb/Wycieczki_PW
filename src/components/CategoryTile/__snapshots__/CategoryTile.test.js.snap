// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryTile should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiCardActionArea-root"
      href="/liceum/kolorowanki/1"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiCardMedia-root"
        style="background-image: url(logo.svg);"
        title="Geometria przestrzenna"
      />
      <div
        class="MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Geometria przestrzenna
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Opis jakiś tam coś tam bla bla
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </a>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
        href="/liceum/kolorowanki/1"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Learn More
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;
